### ISSUE TITLE
Unable to calculate Pearson Type III parameters due to invalid L-moments

### ISSUE THREAD

**Issue by nurkapan** [monocongo/climate_indices#415]
Hi, 
When i try to calculate Spei, i get this error;
2021-02-17  15:10:14 ERROR Unable to calculate Pearson Type III parameters due to invalid L-moments
2021-02-17  15:10:14 ERROR Unable to calculate Pearson Type III parameters due to invalid L-moments
2021-02-17  15:10:14 ERROR Unable to calculate Pearson Type III parameters due to invalid L-moments
2021-02-17  15:10:14 ERROR Unable to calculate Pearson Type III parameters due to invalid L-moments
2021-02-17  15:10:14 ERROR Unable to calculate Pearson Type III parameters due to invalid L-moments
2021-02-17  15:10:15 ERROR Unable to calculate Pearson Type III parameters due to invalid L-moments
2021-02-17  15:10:15 ERROR Unable to calculate Pearson Type III parameters due to invalid L-moments
2021-02-17  15:10:15 ERROR Failed to complete
multiprocessing.pool.RemoteTraceback:
"""
Traceback (most recent call last):
  File "c:\users\nur\anaconda3\envs\indices_env\lib\multiprocessing\pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "c:\users\nur\anaconda3\envs\indices_env\lib\multiprocessing\pool.py", line 48, in mapstar
    return list(map(*args))
  File "c:\users\nur\anaconda3\envs\indices_env\lib\site-packages\climate_indices\__main__.py", line 1398, in _apply_along_axis_double
    computed_array[i, j] = func1d(x[j], y[j], parameters=params["args"])
  File "c:\users\nur\anaconda3\envs\indices_env\lib\site-packages\climate_indices\__main__.py", line 1124, in _spei
    return indices.spei(precips_mm=precips,
  File "c:\users\nur\anaconda3\envs\indices_env\lib\site-packages\climate_indices\compute.py", line 524, in transform_fitted_pearson
    pearson_parameters(
  File "c:\users\nur\anaconda3\envs\indices_env\lib\site-packages\climate_indices\lmoments.py", line 30, in fit
    raise ValueError(message)
ValueError: Unable to calculate Pearson Type III parameters due to invalid L-moments
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "c:\users\nur\anaconda3\envs\indices_env\lib\site-packages\climate_indices\__main__.py", line 1749, in main
    _compute_write_index(kwrgs)
  File "c:\users\nur\anaconda3\envs\indices_env\lib\site-packages\climate_indices\__main__.py", line 1021, in _compute_write_index
    _parallel_process(keyword_arguments["index"],
  File "c:\users\nur\anaconda3\envs\indices_env\lib\site-packages\climate_indices\__main__.py", line 1294, in _parallel_process
    pool.map(_apply_along_axis_double, chunk_params)
  File "c:\users\nur\anaconda3\envs\indices_env\lib\multiprocessing\pool.py", line 364, in map
    return self._map_async(func, iterable, mapstar, chunksize).get()
  File "c:\users\nur\anaconda3\envs\indices_env\lib\multiprocessing\pool.py", line 771, in get
    raise self._value
ValueError: Unable to calculate Pearson Type III parameters due to invalid L-moments
multiprocessing.pool.RemoteTraceback:
"""
Traceback (most recent call last):
  File "c:\users\nur\anaconda3\envs\indices_env\lib\multiprocessing\pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "c:\users\nur\anaconda3\envs\indices_env\lib\multiprocessing\pool.py", line 48, in mapstar
    return list(map(*args))
  File "c:\users\nur\anaconda3\envs\indices_env\lib\site-packages\climate_indices\__main__.py", line 1398, in _apply_along_axis_double
    computed_array[i, j] = func1d(x[j], y[j], parameters=params["args"])
  File "c:\users\nur\anaconda3\envs\indices_env\lib\site-packages\climate_indices\__main__.py", line 1124, in _spei
    return indices.spei(precips_mm=precips,
  File "c:\users\nur\anaconda3\envs\indices_env\lib\site-packages\climate_indices\compute.py", line 524, in transform_fitted_pearson
    pearson_parameters(
  File "c:\users\nur\anaconda3\envs\indices_env\lib\site-packages\climate_indices\lmoments.py", line 30, in fit
    raise ValueError(message)
ValueError: Unable to calculate Pearson Type III parameters due to invalid L-moments
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "c:\users\nur\anaconda3\envs\indices_env\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "c:\users\nur\anaconda3\envs\indices_env\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\nur\anaconda3\envs\indices_env\Scripts\process_climate_indices.exe\__main__.py", line 7, in <module>
  File "c:\users\nur\anaconda3\envs\indices_env\lib\site-packages\climate_indices\__main__.py", line 1749, in main
    _compute_write_index(kwrgs)
  File "c:\users\nur\anaconda3\envs\indices_env\lib\site-packages\climate_indices\__main__.py", line 1021, in _compute_write_index
    _parallel_process(keyword_arguments["index"],
  File "c:\users\nur\anaconda3\envs\indices_env\lib\site-packages\climate_indices\__main__.py", line 1294, in _parallel_process
    pool.map(_apply_along_axis_double, chunk_params)
  File "c:\users\nur\anaconda3\envs\indices_env\lib\multiprocessing\pool.py", line 364, in map
    return self._map_async(func, iterable, mapstar, chunksize).get()
  File "c:\users\nur\anaconda3\envs\indices_env\lib\multiprocessing\pool.py", line 771, in get
    raise self._value
ValueError: Unable to calculate Pearson Type III parameters due to invalid L-moments

I tried both with temperature or pet values and get the same results. What could be reason, can you help please ?

**Comment by monocongo** [comment]
Without looking too closely at this it appears to be an issue with using an out-of-date version of this package. Please follow the installation/usage directions in the documentation. This happens to many users who get the package from drought.gov (an old version, no longer supported) instead of the code from this repository.

TL;DR: install using `pip install climate-indices`

**Comment by nurkapan** [comment]
Actually I already installed it from the documentatiton here and removed and tried again but it did not help. I get this error when i try to calculate spei.

**Comment by Yassminaa** [comment]
I have the same problem, where can find the updated version and how install it?
the documentation refers only to the drought.gov guide.
Thank you

**Comment by nurkapan** [comment]
I followed https://climate-indices.readthedocs.io/en/latest/ steps for installing. I'm not sure if it has anything to do with it, but the data I used was 396 months old, then when I reduced it to 360 months, the program worked.
